nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/clid/ClidObject.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * (C) Copyright 2019 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour Al Kotob
+ */
+
+package org.nuxeo.rest.management.clid;
+
+import java.io.Serializable;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.nuxeo.connect.identity.LogicalInstanceIdentifier;
+import org.nuxeo.connect.identity.LogicalInstanceIdentifier.NoCLID;
+import org.nuxeo.ecm.webengine.model.WebObject;
+import org.nuxeo.ecm.webengine.model.impl.DefaultObject;
+
+/**
+ * @since 11.1
+ */
+@WebObject(type = "clid")
+public class ClidObject extends DefaultObject {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public Object doGet() throws NoCLID {
+        String clid = LogicalInstanceIdentifier.instance().getCLID();
+        Map<String, Serializable> map = new HashMap<>();
```

Comment:
`Map<String,String>` is enough

Reply:
regarding to [`NXP-27483`](https://jira.nuxeo.com/browse/NXP-27483) this feature is only for master. If it's the case then it can be replaced by `Map.of` 
`Map<String, String> clid = Map.of("clid", LogicalInstanceIdentifier.instance().getCLID());` 
wdyt :)

Reply:
is there a raison to use Object as the returned type of this method. i think `Response` can be used wdyt 
@ataillefer any adivce.

Reply:
i am wonedring about what if the reste resource user try/catch bloc and then return an 500 error + Log in the case of the exception NoCLID. i don't know if is there any servlet filter that manage exception in api rest management like the case of Nuxeo @ataillefer wdyt

Reply:
For the fixVersion, sorry it's a mistake (fixed), for now it's only for 10.10, so Java8.

Reply:
Use `Response` yes.

Reply:
About handling the exception, yes that's why a test needs to be added.
Probably need to add the `WebEngineExceptionMapper` to the `ManagementModule`'s singletons to handle the exception. Then in this case I don't think try/catch is necessary since the default error code will be 500 which is fine here from a client's point of view.

The `WebEngineExceptionMapper` will log the 500 error, see https://github.com/nuxeo/nuxeo/blob/master/nuxeo-webengine/nuxeo-webengine-core/src/main/java/org/nuxeo/ecm/webengine/app/WebEngineExceptionMapper.java#L80.

Ultimately, a custom exception mapper for the management REST API could be used, as we did for WOPI: https://github.com/nuxeo/nuxeo/blob/master/addons/nuxeo-wopi/nuxeo-wopi/src/main/java/org/nuxeo/wopi/jaxrs/WOPIExceptionMapper.java

But I guess that for now the webengine one should do.

Reply:
top, it's what i was looking for. i agree with the wrapper no need for try / catch bloc, the  exception mapper will be reponsible of that.
thanks

Reply:
NoClid => 500 💪

Reply:
OK, you get a 500 and an ERROR log server-side because jersey's `ContainerResponse` handles it by default, but it's not really nice and what we expect.

You'll probably get something like
```
ERROR [ContainerResponse] The exception contained within MappableContainerException could not be mapped to a response, re-throwing to the HTTP container
```

in the server logs, and a an ugly HTML error page as a response of the curl call.

Using the `WebEngineExceptionMapper`, you should get:
- A clean error logged server-side by the `WebEngineExceptionMapper`
- Client-side, a JSON reponse with the exception message, cause and stack trace (and still status code 500 of course).